### PR TITLE
Adjust lsp-ui-doc overlay inner padding (when not using a childframe)

### PR DESCRIPTION
**Pull request by apiraino**
Hi,

this is an attempt to study the problem I've reported in #489. I've dug a bit into the code and I think the root of the issue is that `lsp-ui` has not a great support for emacsen that don't use a GUI (either running `emacs -nw` or without the GTK libs compiled in). Note: this is not meant to be a critique, just to frame the issue.

When not using the GUI childframe (i.e. `lsp-ui-doc-use-childframe nil`) it's not clear to me how the overlay is created and if I can interact with it. For example I cannot scroll a long page of documentation:

I understand the technical limits of a non-GUI overlay, I probably couldnt improve a lot here, but this PR is trying to fix at least one issue about the incorrect padding calculated. For some reason the overlay padding from the right border is not enough, therefore each line exceeding the overlay width "bleeds" into the main buffer messing the view (see my issue report).

I'm not converstant with elisp, so I'm very open to any explaination and suggestion to improve this patch.

Thanks for a review and for working on lsp-ui! :+1:

### CODE REVIEW

File: lsp-ui-doc.el [hunk 1]
```diff
@@ -702,12 +702,18 @@ FN is the function to call on click."
 (defvar-local lsp-ui-doc--inline-width nil)
 
 (defun lsp-ui-doc--inline-window-width nil
+  "Calculate inner padding for the child frame showing the documentation"
+  (if (not (bound-and-true-p lsp-ui-doc-use-childframe))
+      (setq additional-width-spacing 8)
```
Comment (apiraino) [inline lsp-ui-doc.el:707]: the value of `8` here is a complete guess of what the minimum good value to fix the problem is. Same reasoning for the default value of `1` that lsp-ui-doc is already setting as an alternate value.

File: lsp-ui-doc.el [hunk 2]
```diff
@@ -702,12 +702,18 @@ FN is the function to call on click."
...
+
   (- (min (window-text-width)
           (window-body-width))
      (if (bound-and-true-p display-line-numbers-mode)
          (+ 2 (line-number-display-width))
-       0)
-     1))
+         0)
+     additional-width-spacing)
+  )
```
Comment (jcs090218) [inline lsp-ui-doc.el:716]: Can you use `let` operator for local variable `additional-width-spacing`? And don't put `)` on a newline (L716). Thanks!

File: lsp-ui-doc.el (hunk 1 above)
Comment (jcs090218) [reply lsp-ui-doc.el:707]: Make sense to me.

File: lsp-ui-doc.el (hunk 2 above)
Comment (apiraino) [reply lsp-ui-doc.el]: done

**Comment by jcs090218** [comment]
Not sure if you can use `let` operator this way. Let's try this.

```el
(defun lsp-ui-doc--inline-window-width nil
  "Calculate inner padding for the child frame showing the documentation"
  (let ((additional-width-spacing (if (bound-and-true-p lsp-ui-doc-use-childframe)
                                      8 1)))
    (- (min (window-text-width) (window-body-width))
       (if (bound-and-true-p display-line-numbers-mode)
           (+ 2 (line-number-display-width)) 0)
       additional-width-spacing)))
```

This is a much simplify version of function `lsp-ui-doc--inline-window-width`.

**Comment by apiraino** [comment]
embarassing how little elisp I know :smile: 

and now the CI pass! thank you so much @jcs090218 for your help

**Comment by jcs090218** [comment]
@apiraino No problem! Glad to help out!

Can someone test this patch? @sebastiencs @brotzeit

**Comment by apiraino** [comment]
Unfortunately I think there are still issues with the overlay width. Simply setting a larger value (8 instead of 1) is not useful. I am testing with more code and for example the overlay breaks with long function signatures (example +100 characters).

I welcome a review and suggestions, but I'd keep this PR in standby until I figure out a better approach (example: by dynamically calculating the width).

**Comment by jcs090218** [comment]
I might try to avoid magic number like `8` or `1`. Calculating the padding by longest text lines sounds like a good approach. ;)

**Comment by apiraino** [comment]
I think I'll close this and clean up some mess I've left. I couldn't get around to fixing this and I can't reproduce it anymore after migrating to another Emacs config

**Pull request closed without merging**